src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -91,4 +102,27 @@ public Book findOne(@PathVariable Long id) {
                         () -> new ResponseStatusException(HttpStatus.NOT_FOUND,
                                 Constants.NOT_FOUND));
     }
+
+    /**
+     * This method find a Book by ISBN, if not return find in a external service and persist in data
+     * base
+     *
+     * @param isbn
+     * @return
+     */
+    @GetMapping()
+    public ResponseEntity findByIsbn(@RequestParam("isbn") String isbn) {
+        Book book = bookRepository.findByIsbn(isbn).orElse(null);
+        if (book == null) {
```

Comment:
You could use Optional to handle this without a null comparison.
Something like:
```
Book book = bookRepository.findByIsbn(isbn);
if (book.isPresent()) {
  return book.get();
}
// else
.....
```